File: tests/parsers/samples/code/dart/simple.dart
import 'complex.dart';

Map<String, double> runSimple() {
  final analyzer = ComplexMetrics.buildDefault();
  final rms = analyzer.rootMeanSquare([1.0, 2.0, 3.0]);
  final baseline = analyzer.distance(10, 3);
  return {
    'baseline': baseline,
    'rms': rms,
  };
}

String summarize() {
  final metrics = runSimple();
  return "baseline=${metrics['baseline']?.toStringAsFixed(2)}, rms=${metrics['rms']?.toStringAsFixed(2)}";
}

File: tests/parsers/samples/code/rb/simple.rb
require_relative 'complex'

module SimpleMetrics
  module_function

  def run_simple
    {
      baseline: ComplexMetrics.distance(10, 3, 0.8),
      rms: ComplexMetrics.root_mean_square([1.0, 2.0, 3.0])
    }
  end

  def summarize
    metrics = run_simple
    format('baseline=%.2f, rms=%.2f', metrics[:baseline], metrics[:rms])
  end
end
